src/main/java/network/brightspots/rcv/CastVoteRecord.java

```diff
@@ -44,29 +45,50 @@
   private final String suppliedId;
   // which precinct this ballot came from
   private final String precinct;
-  // contest associated with this CVR
-  private Integer contestId;
-  // tabulatorId parsed from Dominion CVR data
-  private Integer tabulatorId;
-  // batchId parsed from Dominion CVR data
-  private Integer batchId;
   // container for ALL CVR data parsed from the source CVR file
   private final List<String> fullCvrData;
   // records winners to whom some fraction of this vote has been allocated
   private final Map<String, BigDecimal> winnerToFractionalValue = new HashMap<>();
+  // If CVR CDF output is enabled, we store the necessary info here: for each round, the list of
+  // candidates this ballot is counting toward (0 or 1 in a single-seat contest; 0 to n in a
+  // multi-seat contest because of fractional vote transfers), and how much of the vote each is
+  // getting. As a memory optimization, if the data is unchanged from the previous round, we don't
+  // add a new entry.
+  private final Map<Integer, List<Pair<String, BigDecimal>>> cdfSnapshotData = new HashMap<>();
   // map of round to all candidates selected for that round
   // a set is used to handle overvotes
   SortedMap<Integer, Set<String>> rankToCandidateIds;
+  // contest associated with this CVR
+  private Integer contestId;
+  // tabulatorId parsed from Dominion CVR data
+  private Integer tabulatorId;
+  // batchId parsed from Dominion CVR data
+  private Integer batchId;
+  // ballotTypeId parsed from Dominion CVR data
+  private Integer ballotTypeId;
```

Comment:
IntelliJ points out that these four variables are never actually accessed anywhere. I assume that's just because we haven't yet built the part that uses them to calculate results? Which issue would this be tied to?

Reply:
Yeah, these will be read and written as part of the new .csv exported CVR file which is #404.